app/models/user.rb

```diff
@@ -10,8 +10,9 @@ class User < ApplicationRecord
 
   def self.new_with_session(params, session)
     super.tap do |user|
-      if data = session["devise.facebook_data"] && session["devise.facebook_data"]["extra"]["raw_info"]
-        user.email = data["email"] if user.email.blank?
+      data = session['devise.facebook_data']
+      if data && session['devise.facebook_data']['extra']['raw_info']
```

Comment:
```suggestion
      if data && data['extra']['raw_info']
```
I just wonder what connection has `extra/raw_info` with `email`